src/components/Pressable.tsx

```diff
@@ -0,0 +1,286 @@
+      setChildrenProp(props.children);
+    }
+  };
+
+  const hoverGesture = Gesture.Hover()
+    .onBegin((event) => {
+      setTimeout(
+        () => props.onHoverIn?.(adaptStateChangeEvent(event)),
+        props.delayHoverIn ?? DEFAULT_HOVER_DELAY
+      );
+    })
+    .onEnd((event) => {
+      setTimeout(
+        () => props.onHoverOut?.(adaptStateChangeEvent(event)),
+        props.delayHoverOut ?? DEFAULT_HOVER_DELAY
+      );
+    });
+
+  const touchGesture = Gesture.Manual()
+    .onTouchesDown((event) => {
+      pressableRef.current?.measure((_x, _y, width, height) => {
+        if (
+          !isTouchWithinInset(event.changedTouches.at(-1), normalizedHitSlop, {
+            width,
+            height,
+          }) ||
+          isPressedDown.current
+        ) {
+          return;
+        }
+
+        props.onPressIn?.(adaptTouchEvent(event));
+
+        isPressCallbackEnabled.current = true;
+        setPressedState(true);
+      });
+    })
+    .onTouchesUp((event) => {
+      if (
+        !isPressedDown.current ||
+        event.allTouches.length > event.changedTouches.length
+      ) {
+        return;
+      }
+
+      props.onPressOut?.(adaptTouchEvent(event));
+
+      if (isPressCallbackEnabled.current) {
+        props.onPress?.(adaptTouchEvent(event));
+      }
+
+      setPressedState(false);
+    })
+    .onTouchesCancelled((event) => {
+      if (
+        !isPressedDown.current ||
+        event.allTouches.length > event.changedTouches.length
+      ) {
+        return;
+      }
+
+      // original triggers onPressOut on cancel, but not onPress
+      props.onPressOut?.(adaptTouchEvent(event));
+
+      setPressedState(false);
+    });
+
+  const rippleGesture = Gesture.Native();
+
+  pressGesture.minDuration(props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION);
+  pressGesture.minDuration(props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION);
+
+  const appliedHitSlop = addInsets(
+    normalizedHitSlop,
+    normalizedPressRetentionOffset
+  );
+
+  // this hitSlop block is required by android
+  touchGesture.hitSlop(appliedHitSlop);
+  pressGesture.hitSlop(appliedHitSlop);
+  hoverGesture.hitSlop(appliedHitSlop);
+  rippleGesture.hitSlop(normalizedHitSlop);
+
+  touchGesture.shouldCancelWhenOutside(true);
+  pressGesture.shouldCancelWhenOutside(true);
+  hoverGesture.shouldCancelWhenOutside(true);
+
+  touchGesture.enabled(props.disabled !== true);
+  pressGesture.enabled(props.disabled !== true);
+  hoverGesture.enabled(props.disabled !== true);
+
+  touchGesture.runOnJS(true);
+  pressGesture.runOnJS(true);
+  hoverGesture.runOnJS(true);
+
+  const gesture = Gesture.Simultaneous(
+    hoverGesture,
+    pressGesture,
+    touchGesture,
+    rippleGesture
+  );
+
+  useEffect(() => {
+    setPressedState(props.testOnly_pressed ?? false);
+  }, []);
+
+  const defaultRippleColor = props.android_ripple ? undefined : 'transparent';
+
+  return (
+    <GestureDetector gesture={gesture}>
+      <RNButton
```

Comment:
I think it should also receive `enabled` prop

Reply:
Done